addons/account/wizard/accrued_orders.py

```diff
@@ -188,15 +192,23 @@ def _ellipsis(string, size):
                         amount_currency = order_line.untaxed_amount_to_invoice
                         fnames = ['qty_to_invoice', 'untaxed_amount_to_invoice', 'qty_invoiced', 'qty_delivered', 'invoice_lines']
                         label = _('%s - %s; %s Invoiced, %s Delivered at %s each', order.name, _ellipsis(order_line.name, 20), order_line.qty_invoiced, order_line.qty_delivered, formatLang(self.env, order_line.price_unit, currency_obj=order.currency_id))
-                    values = _get_aml_vals(order, amount, amount_currency, account.id, label=label)
+                    values = _get_aml_vals(order, amount, amount_currency, account.id, label=label, analytic_distribution=order_line.analytic_distribution)
                     move_lines.append(Command.create(values))
                     total_balance += amount
                 # must invalidate cache or o can mess when _create_invoices().action_post() of original order after this
                 order.order_line.invalidate_model(fnames)
 
         if not self.company_id.currency_id.is_zero(total_balance):
             # globalized counterpart for the whole orders selection
-            values = _get_aml_vals(orders, -total_balance, 0.0, self.account_id.id, label=_('Accrued total'))
+            analytic_distribution = {}
+            total = sum(order.amount_total for order in orders)
+            for order in orders:
+                ratio = order.amount_total / total
+                if not order.order_line.analytic_distribution:
+                    continue
+                for k, v in order.order_line.analytic_distribution.items():
```

Comment:
You can have multiple order lines, you can't get a field of a recordset of `len > 1` like this
```suggestion
            for line in orders.order_line:
                ratio = line.price_total / total
                if not line.analytic_distribution:
                    continue
                for k, v in line.analytic_distribution.items():
```

Reply:
Right, I've change it. Just waiting for some validation from tsb on a more complex example.